compiler/src/main/java/dagger/internal/codegen/BindingGraph.java

```diff
@@ -210,14 +215,26 @@ private BindingGraph create(
         explicitBindingsBuilder.add(provisionBindingFactory.forComponent(componentDependency));
         List<ExecutableElement> dependencyMethods =
             ElementFilter.methodsIn(elements.getAllMembers(componentDependency));
+        HashMultimap<Key, Equivalence.Wrapper<TypeMirror>> componentMethodType = HashMultimap.create();
```

Comment:
```
@Component(dependencies = Dep.class)
interface C {
  A a();
}

class Dep {
  A a() { return new A("hello"); }
  A b() { return new A("world"); }
}
```

How do we know which method on `Dep` to call? Without knowing anything about the type, we have to create a binding for each method.

Your issue is different, since from the perspective of the component taking dependencies, there's only one method, but the annotation processing API is giving two.